src/mpool.c

```diff
@@ -120,12 +120,7 @@ void *mpool_alloc(mpool_t *mp)
 
 void *mpool_calloc(mpool_t *mp)
 {
-    if (!mp->chunk_count && !(mpool_extend(mp)))
-        return NULL;
-
-    char *ptr = (char *) mp->free_chunk_head + sizeof(memchunk_t);
-    mp->free_chunk_head = mp->free_chunk_head->next;
-    mp->chunk_count--;
+    char *ptr = mpool_alloc(mp);
```

Comment:
The mpool_alloc(mp) may return NULL and assign to `ptr`, and this modification lacks checking whether `ptr` is NULL before `memset`. Upon my examination, this situation lead to segmentation faults when `ptr` is NULL. Therefore, this modificatio needs to add another checking for `ptr` before `memset`.

Reply:
Thank you for the suggestion, @qwe661234 . I have checked if `ptr` is not NULL using `if(ptr)` to ensure that executing `memset` will not cause segmentation faults.